### PR TITLE
feat: assistant with openAI

### LINKED ISSUE #162: Develop the conversational UI for interacting with the Assistant, focusing on natural language understanding and user-friendly design.

### PR DESCRIPTION
**Pull request by pixelass**
## Motivation

adds a simple chat assistant

# Issues closed 

closes #162

### CODE REVIEW

File: src/client/apps/assistant/constants.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const APP_ID = "assisitant";
```
Comment (TimPietrusky) [inline src/client/apps/assistant/constants.ts]:
```suggestion
export const APP_ID = "assistant";
```

File: src/client/atoms/icons/chat-gpt.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import type { SvgIconProps } from "@mui/material/SvgIcon";
+import SvgIcon from "@mui/material/SvgIcon";
+
+export default function ChatGPTIcon(properties: SvgIconProps) {
```
Comment (TimPietrusky) [inline src/client/atoms/icons/chat-gpt.tsx:4]: Why are we using the ChatGPT Icon here? Isn't this a general purpose assistant that could feature any model?

File: src/electron/helpers/ipc/sdk/openai.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+		const channel = `${appId}:${APP_MESSAGE_KEY}`;
+		const chatId = message.payload.id;
+		let chat = openAIStore.get(chatId, {
+			label: "New Chat",
+			messages: [],
+			model: "gpt-3.5-turbo",
+		});
+
+		switch (message.action) {
```
Comment (TimPietrusky) [inline src/electron/helpers/ipc/sdk/openai.ts:48]: Can you please add a comment for each action, so that it's clear what the action is doing?

**Review by TimPietrusky — CHANGES_REQUESTED** [review]
Looks super nice already, just a few minor things.

File: src/client/atoms/icons/chat-gpt.tsx (hunk 2 above)
Comment (pixelass) [reply src/client/atoms/icons/chat-gpt.tsx:4]: yes, other providers will then have a different icon. 
You can switch models, in a chat. But right now there are only GPT models

**Review by TimPietrusky — APPROVED** [review]
LGTM, LGTM!

**Pull request merged**